core/schema/module.go

```diff
@@ -95,140 +92,84 @@ func (s *moduleSchema) Install() {
 	}.Install(s.dag)
 
 	dagql.Fields[*core.ModuleSource]{
-		dagql.Func("contextDirectory", s.moduleSourceContextDirectory).
-			Doc(`The directory containing everything needed to load and use the module.`),
-
-		dagql.Func("withContextDirectory", s.moduleSourceWithContextDirectory).
-			Doc(`Update the module source with a new context directory. Only valid for local sources.`).
-			ArgDoc("dir", `The directory to set as the context directory.`),
-
-		dagql.Func("directory", s.moduleSourceDirectory).
-			Doc(`The directory containing the module configuration and source code (source code may be in a subdir).`).
-			ArgDoc(`path`, `The path from the source directory to select.`),
-
-		dagql.Func("sourceRootSubpath", s.moduleSourceRootSubpath).
-			Doc(`The path relative to context of the root of the module source, which contains dagger.json. It also contains the module implementation source code, but that may or may not being a subdir of this root.`),
-
-		dagql.Func("sourceSubpath", s.moduleSourceSubpath).
-			Doc(`The path relative to context of the module implementation source code.`),
+		Syncer[*core.ModuleSource]().
+			Doc(`TODO`),
 
 		dagql.Func("withSourceSubpath", s.moduleSourceWithSourceSubpath).
 			Doc(`Update the module source with a new source subpath.`).
 			ArgDoc("path", `The path to set as the source subpath.`),
 
-		dagql.Func("moduleName", s.moduleSourceModuleName).
-			Doc(`If set, the name of the module this source references, including any overrides at runtime by callers.`),
-
-		dagql.Func("moduleOriginalName", s.moduleSourceModuleOriginalName).
-			Doc(`The original name of the module this source references, as defined in the module configuration.`),
-
 		dagql.Func("withName", s.moduleSourceWithName).
 			Doc(`Update the module source with a new name.`).
 			ArgDoc("name", `The name to set.`),
 
-		dagql.NodeFunc("dependencies", s.moduleSourceDependencies).
-			Doc(`The effective module source dependencies from the configuration, and calls to withDependencies and withoutDependencies.`),
+		dagql.Func("withSDK", s.moduleSourceWithSDK).
+			Doc(`Update the module source with a new SDK.`).
+			ArgDoc("sdk", `The SDK to set.`),
+
+		dagql.Func("withInit", s.moduleSourceWithInit).
+			Doc(`Sets module init arguments`).
+			ArgDoc("merge", `Merge module dependencies into the current project's`),
```

Comment:
Hm, another while we're here... @marcosnils any objection to removing this now? I remember we changed the default and left this in here as a way to get the old behavior.

But we never documented it, and I don't know that anyone relies on it? Since we're doing breaking things anyways here, I figure it probably wouldn't hurt to just remove it and simplify the logic changed in https://github.com/dagger/dagger/pull/8095.

Reply:
I removed it from the CLI. There's still a need for a flag on `codegen` because I think the go sdk (non module) build actually requires the behavior, but it's 100% internal now.